components/scream/src/share/tests/workspace_tests.cpp

```diff
@@ -0,0 +1,316 @@
+#include "catch2/catch.hpp"
+
+#include "share/wsm.hpp"
+#include "share/util/scream_kokkos_utils.hpp"
+
+namespace unit_test {
+
+using namespace scream;
+
+struct UnitWrap {
+
+template <typename D=DefaultDevice>
+struct UnitTest {
```

Comment:
It may be worth following the pattern in this file (most of the test impl living in a static function of a struct with a device template) for most of our tests. It makes it very easy to test multiple device types and allows reuse of the common Kokkos types.